lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2228,15 +2248,31 @@ class HSADevice final : public KalmarDevice
     UnpinnedCopyEngine::CopyMode  copy_mode;
 
     // Creates or steals a rocrQueue and returns it in theif->rocrQueue
-    void createOrstealRocrQueue(Kalmar::HSAQueue *thief) {
+    void createOrstealRocrQueue(Kalmar::HSAQueue *thief, queue_priority priority = priority_normal) {
         RocrQueue *foundRQ = nullptr;
 
+        // Map Kalmar::enums:queue_priority to rqIndex which is used for indexing into rocrQueues
+        int rqIndex = 1;
+        switch (priority) {
```

Comment:
I think it would be more clear to use `priority` to index into rocrQueues rather than creating another index `rqIndex`

Reply:
```priority``` is an enum. Using an enum for index arrays is usually a bad practice. Especially if we add additional enum values later or change the associated values later. Hence the mapping. However, if you prefer using the enum for indexing, I can clean this up.

Reply:
I don't think creating another set of ids that needs to be maintained manually would be a better solution to the issue you described.  The enumerators have defined values so just need documentation that we use them to get to the right queue pool.  You could use a map if the value of enumerators are non-contiguous but that seems overkill